### PR TITLE
Homepage blog slice

### PR DESCRIPTION
**Pull request by asavin**
giphy

### Motivation
https://trello.com/c/cqCPTtL1/169-website-honestly-homepage-blog-slice

### Test plan

zeplin

- Check featured blog slice on all devices

### Pre-flight checklist
- [x] Has been design reviewed
- [x] Add a gif

### CODE REVIEW

**Comment by asavin** [comment]
To be fair I'll need https://github.com/redbadger/website-honestly/pull/45 first to refactor this one properly.

**Comment by asavin** [comment]
Received 💚 from design

File: site/pages/home/blog-slice/blog-entry/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    </li>
+  );
+};
+
+BlogEntry.propTypes = {
+  category: PropTypes.string,
+  title: PropTypes.string,
+  url: PropTypes.string,
+  authorName: PropTypes.string,
+  authorTitle: PropTypes.string,
```
Comment (lpil) [inline site/pages/home/blog-slice/blog-entry/index.js]: None of these are marked as required. What should happen if they are not passed?

**Comment by AndreiDanielIonescu** [comment]
Several issues found while testing:
 - on Firefox, when you click to a link/go to the link/return a rectangle remains over the link
 - on Safari there is no underline while hovering over the link
 - on Safari IOS based the underline remains for the link you visited

**Comment by asavin** [comment]
@lpil the current tests explode if any of the components fail to render properly.

**Comment by lpil** [comment]
Yes, but for only one path. This is an extremely high level integration path.

This component has several different possible inputs, are they worth testing? If not as only one is used why are there multiple inputs?

If we decide to make all the fields mandatory how is this to be translated over to the variable data from BB at a later date?

**Comment by asavin** [comment]
@AndreiDanielIonescu On El Capitan Safari hover issue:
* If you scroll down and go straight to the blog slice and hover over links, underline on hover doesn't appear
* If you hover over navigation bar links first - underline appears - and then scroll to blog slice and hover over links - it works and underline appears on hover

I'd say this is a minor Safari issue, which is not affecting any functionality. According to Internet it also might be fixed with the more recent Mac OS update - something probably worth testing.

File: site/pages/home/blog-slice/blog-entry/index.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <li>
+      <a
+        href={url}
+        className={styles.link}
+      >
+        <h3 className={styles.category}>{category}</h3>
+        <div className={styles.linkEntry}>
+          <div className={styles.linkTitle}>
+            <p>{title}</p>
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/blog-entry/index.js]: Perhaps more thinking for later. Please consider semantics for accessibility

File: site/pages/home/blog-slice/blog-entry/style.css [hunk 3]
```diff
@@ -0,0 +1,70 @@
+@value white, black, red, linesOnWhite from "../../../../css/_colors.css";
+
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/blog-entry/style.css]: Remove space

File: site/pages/home/blog-slice/blog-entry/style.css [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  color: red;
+}
+
+.authorTitle {
+  display: none;
+}
+
+@media (hover:hover) {
+  .linkTitle p:hover {
+    border-bottom: 2px solid red;
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/blog-entry/style.css]: Check if this is a variable

File: site/pages/home/blog-slice/index.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import styles from './style.css';
+import BlogEntry from './blog-entry';
+
+const homepageBlogSlice = () => {
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/index.js]: just BlogSlice

File: site/pages/home/blog-slice/index.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <section className={styles.homepageBlogSlice}>
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/index.js]: Minus the homepage

File: site/pages/home/blog-slice/style.css [hunk 7]
```diff
@@ -0,0 +1,51 @@
+@value white, black, red, linesOnWhite from "../../../css/_colors.css";
+
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/style.css]: Remove space. I will do it

File: site/pages/home/blog-slice/style.css [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  composes: fontL from "../../../css/typography/_fonts.css";
+  composes: serif from "../../../css/typography/_fonts.css";
+}
+
+.blogLink {
+  composes: primaryButton from "../../../css/_links.css";
+}
+
+.blogLinkList {
+  margin: 30px 0 30px 0;
```
Comment (AndrewBestbier) [inline site/pages/home/blog-slice/style.css]: If you are feeling motivated

File: site/pages/home/blog-slice/blog-entry/style.css (hunk 4 above)
Comment (asavin) [inline site/pages/home/blog-slice/blog-entry/style.css]: There's a class, but it has also text colour which is wrong

**Pull request merged**